src/Gom/main.js

```diff
@@ -0,0 +1,159 @@
+import GObject from "gi://GObject";
+import Gom from "gi://Gom";
+import Gtk from "gi://Gtk";
+import Adw from "gi://Adw";
+import GLib from "gi://GLib";
+import Gio from "gi://Gio";
+
+const ItemClass = GObject.registerClass(
+  {
+    GTypeName: "Item",
+    Properties: {
+      id: GObject.ParamSpec.int(
+        "id",
+        "ID",
+        "An ID",
+        GObject.ParamFlags.READWRITE | GObject.ParamFlags.CONSTRUCT,
+        0,
+        GLib.MAXINT32,
+        0,
+      ),
+      url: GObject.ParamSpec.string(
+        "url",
+        "URL",
+        "A URL",
+        GObject.ParamFlags.READWRITE | GObject.ParamFlags.CONSTRUCT,
+        "",
+      ),
+    },
+  },
+  class ItemClass extends Gom.Resource { },
+);
+
+let adapter, repository;
+
+function initDatabase() {
+  adapter = new Gom.Adapter();
+  adapter.open_sync(":memory:");
+  repository = new Gom.Repository({ adapter: adapter });
+
+  // Set up table and primary key
+  ItemClass.set_table("items");
+  ItemClass.set_primary_key("id");
+
+  // Perform automatic migration
+  repository.automatic_migrate_sync(1, [ItemClass]);
+}
+
+function closeDatabase() {
```

Comment:
Once the database is closed, opening it again would need an accompanying closed signal since the adapter leaks otherwise. Are we thinking of a simple button to close the database to avoid the alert?